src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1273,15 +1273,29 @@ private AttributeSemanticModel CreateModelForAttribute(Binder enclosingBinder, A
             AliasSymbol aliasOpt;
             var attributeType = (NamedTypeSymbol)enclosingBinder.BindType(attribute.Name, BindingDiagnosticBag.Discarded, out aliasOpt).Type;
 
+            // For attributes where a nameof could introduce some type parameters, we need to track the attribute target
+            Symbol attributeTarget = GetAttributeTargetForExtraTypeParameters(attribute.Parent.Parent);
+
             return AttributeSemanticModel.Create(
                 this,
                 attribute,
                 attributeType,
                 aliasOpt,
+                attributeTarget,
                 enclosingBinder.WithAdditionalFlags(BinderFlags.AttributeArgument),
                 containingModel?.GetRemappedSymbols());
         }
 
+        Symbol GetAttributeTargetForExtraTypeParameters(SyntaxNode targetSyntax)
+        {
+            return targetSyntax switch
+            {
+                MethodDeclarationSyntax methodDeclaration => GetDeclaredMemberSymbol(methodDeclaration),
+                LocalFunctionStatementSyntax localFunction => GetMemberModel(localFunction).GetDeclaredLocalFunction(localFunction),
```

Comment:
>GetMemberModel(localFunction) [](http://example.com/codeflow?start=62&length=29)

Is this never going to fail? Perhaps conditional access?

Reply:
I wasn't able to hit it, but that seems safer. Thanks